modules/project-factory/variables.tf

```diff
@@ -299,6 +324,19 @@ variable "factories_data" {
           iam_sa_roles           = optional(map(list(string)), {})
           iam_storage_roles      = optional(map(list(string)), {})
         })), {})
+        templates = optional(object({
+          workflow = optional(object({
+            plan = object({
+              service_account = string
+            })
+            apply = object({
+              service_account = string
+            })
+          }))
+          provider_files = optional(list(object({
```

Comment:
Thinking about FAST here, where we're already generating the provider files and storing them in the FAST-level output bucket/location: I would remove this from the variables and always generate provider files if the output bucket is present, or an existing bucket name is passed in. FAST would then simply instruct the project factory to use the provided bucket for the files. There should also be an option to generate them locally.

Reply:
Addendum: I would keep a single `templates.provider`, and use a default one if nothing is passed in and an output bucket or bucket details are present.

For workflows, I would make it a map and not an object, with `workflow_file` and `service_account` attributes for each element.

Reply:
I've updated this.  Please take a look and let me know what you think.